src/array.ts

```diff
@@ -401,6 +411,12 @@ Array.prototype.count = function <T>(this: T[], f: (el: T, index: number, array:
   return this.reduce((count, ...args) => (f(...args) ? count + 1 : count), 0);
 };
 
+Array.prototype.unanimous = function <T>(this: T[], f: (el: T) => unknown): boolean {
+  const [first, ...rest] = this;
+  const comparator = f(first);
```

Comment:
Hm, seems like this would NPE a comparator like `.unanimous(a => a.firstName)` b/c `a` would be `undefined`?

Reply:
NPE?

Reply:
NullPointerException, `undefined.firstName` will fail, granted TypeError in JS terms, but seems like TypeError can happen for more than just NPE? Maybe? Not sure. Admittedly used NPE as a colloquialism.

Reply:
The same thing can happen in `.map(a => a.firstName)` or `forEach` or all the rest. This prototype shouldn't be the first one that has to go out of its way to handle that case.

Per my other comment, I tried to make sure TypeScript would recognize that case and force the user to deal with it in the `fn` they pass in but in _this_ repo for whatever reason it does not recognize `[1,2,3,undefined,5]` as `(number | undefined)[]`

Reply:
Yeah, but TS will think:

```
const authors = [{ firstName: "a" }, { firstName: "b" }];
```

Is a `Array<{ firstName: string }>`, so it will allow:

```
authors.everyHasSame(a => a.firstName)
```

But if at runtime, `authors = []`, it's still typed as `Array<Author>` (not `Author<Author | undefined>)`, then `a => a.firstName` is legal / will compile, but you're going to cause `undefined.firstName` by invoking `fn(undefined)`.

Per your mention of "this is how `.map` works", if `authors = []`, the `map` internal implement never calls the lambda `fn`, but currently you're _always_ doing that, to probe the first entry, which is why this is different.

Per not getting TS errors, try adding `strict: true` to tsconfig.

Reply:
Gotcha. Will check it out 👍🏻